### PR TITLE
Added ability to modify memory in simulator

### PR DESCRIPTION
**Pull request by alexhrv**

### CODE REVIEW

**Comment by alexhrv** [comment]
MemDumpCommand.cc had 2 additional check at lines 203 and 214 and I wasn't sure should I add them also

**Comment by pjaaskel** [comment]
Please add some basic smoke tests on top of ttasim under systemtest/codesign/ttasim. Use the new tcetest*.sh format. I don't remember if it's documented anywere, so just ask Aleksi for an introduction.

File: tce/src/applibs/Simulator/MemWriteCommand.cc [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/*
+    Copyright (c) 2002-2009 Tampere University of Technology.
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/MemWriteCommand.cc]: Updated copyright year.

File: tce/src/applibs/Simulator/MemWriteCommand.cc [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+#include "StringTools.hh"
+#include "Address.hh"
+#include "NullAddressSpace.hh"
+#include "SimValue.hh"
+#include "Conversion.hh"
+#include "FileSystem.hh"
+#include <iostream>
+#include <fstream>
+
+/**
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/MemWriteCommand.cc]: You can remove these trivial comments from the constructor and destructor. We don't add those anymore.

File: tce/src/applibs/Simulator/MemWriteCommand.cc [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+ *
+ * @param arguments Memory address to be written, file to read from, read size.
+ * @return True in case simulation is initialized and arguments are ok.
+ * @exception NumberFormatException Is never thrown by this command.
+ */
+bool 
+MemWriteCommand::execute(const std::vector<DataObject>& arguments)
+    throw (NumberFormatException) {
+
+    const int argumentCount = arguments.size() - 1;
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/MemWriteCommand.cc]: Is this a copy-paste from MemDumpCommand? Please extract it to another method and call it instead.

File: tce/src/applibs/Simulator/MemWriteCommand.cc [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        writeAddress++;
+        readSize--;
+    }
+    memory->advanceClock();
+    inputFile.close();
+
+    return true;
+}
+
+/**
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/MemWriteCommand.cc]: Also a trivial comment that can be now omitted.

File: tce/src/applibs/Simulator/MemWriteCommand.hh [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/*
+    Copyright (c) 2002-2009 Tampere University of Technology.
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/MemWriteCommand.hh]: Copyright year.

File: tce/src/applibs/Simulator/MemDumpCommand.cc [hunk 6]
```diff
@@ -136,15 +136,8 @@ MemDumpCommand::execute(const std::vector<DataObject>& arguments)
...
-            try {
-                const TTAProgram::Address& parsedAddress =
-                    parseDataAddressExpression(addressString);
-                if (&parsedAddress.space() != 
-                    &TTAMachine::NullAddressSpace::instance()) {
-                    addressSpaceName = parsedAddress.space().name();
-                }
-                newDisplayedAddress = parsedAddress.location();
-            } catch(const IllegalParameters& n) {
+            if (!setMemoryAddress(addressString, addressSpaceName,
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/MemDumpCommand.cc]: Check the indentation of the parameter list.

File: tce/src/applibs/Simulator/SimControlLanguageCommand.cc [hunk 7]
```diff
@@ -722,6 +722,75 @@ SimControlLanguageCommand::verifyBreakpointHandles(
     return true;
 }
 
+/**
+ * Sets corresponding memory address and address space given by address string.
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/SimControlLanguageCommand.cc]: Should the word 'parse' used here? Parses the address tring to corresponding memory address and address space or similar.

File: tce/src/applibs/Simulator/SimControlLanguageCommand.cc [hunk 8]
```diff
@@ -722,6 +722,75 @@ SimControlLanguageCommand::verifyBreakpointHandles(
...
+ *
+ * @param addressString Memory address.
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/SimControlLanguageCommand.cc]: Memory address command string to parse.

File: tce/src/applibs/Simulator/SimControlLanguageCommand.cc [hunk 9]
```diff
@@ -722,6 +722,75 @@ SimControlLanguageCommand::verifyBreakpointHandles(
...
+ * @param addressSpaceName Address space name to be set.
+ * @param memoryAddress Memory address to be set.
+ * @return true on success.
+ */
+bool
+SimControlLanguageCommand::setMemoryAddress(
+    const std::string& addressString,
+    std::string& addressSpaceName,
```
Comment (pjaaskel) [inline tce/src/applibs/Simulator/SimControlLanguageCommand.cc]: Could it also find the Memory also or is the name needed?

**Pull request merged**